pkg/deployment/deployment.go

```diff
@@ -315,6 +309,54 @@ func (builder *Builder) WithServiceAccountName(serviceAccountName string) *Build
 	return builder
 }
 
+// WithVolume sets the Volume on the deployment definition.
+func (builder *Builder) WithVolume(volume *coreV1.Volume) *Builder {
```

Comment:
```suggestion
func (builder *Builder) WithVolume(volume *coreV1.Volume, mountPath string) *Builder {
```

Reply:
we don't need mount path, it is already inside volume variable:
```go
MountPath: volume.HostPath.Path
```

Reply:
These values aren't equivalent:
https://pkg.go.dev/k8s.io/api/core/v1#VolumeMount
https://pkg.go.dev/k8s.io/api/core/v1#Volume
By using volume.HostPath.Path as the mountPath you are effectively limiting this method to only working with HostPath volumes.